src/containers/votingTerminal/index.tsx

```diff
@@ -105,15 +110,26 @@ export const VotingTerminal: React.FC<VotingTerminalProps> = ({
   const {api: provider} = useProviders();
   const {t} = useTranslation();
   const {network} = useNetwork();
+  const fetchPastVotingPower = usePastVotingPowerAsync();
 
   useEffect(() => {
     // fetch avatar fpr each voter
     async function fetchEns() {
       const response = await Promise.all(
         voters.map(async voter => {
           const wallet = await Web3Address.create(provider, voter.wallet);
+          let balance;
+          if (daoToken?.address && wallet.address) {
+            balance = await fetchPastVotingPower({
+              tokenAddress: daoToken.address as string,
+              address: wallet.address as string,
+              blockNumber: blockNumber as number,
+            });
+          }
           return {
             ...voter,
+            tokenAmount: balance ? formatUnits(balance, 18) : voter.tokenAmount,
```

Comment:
We probably need to use the decimals set on the daoToken variable:

```suggestion
            tokenAmount: balance ? formatUnits(balance, daoToken.decimals) : voter.tokenAmount,
```